Source/Scene/Points3DTileContent.js

```diff
@@ -340,12 +423,80 @@ define([
         var positions = parsedContent.positions;
         var colors = parsedContent.colors;
         var normals = parsedContent.normals;
+        var batchIds = parsedContent.batchIds;
         var isQuantized = parsedContent.isQuantized;
         var isOctEncoded16P = parsedContent.isOctEncoded16P;
+        var styleableProperties = parsedContent.styleableProperties;
         var isTranslucent = content._isTranslucent;
 
         var hasColors = defined(colors);
         var hasNormals = defined(normals);
+        var hasBatchIds = defined(batchIds);
+
+        var batchTable = content.batchTable;
+        var hasBatchTable = defined(batchTable);
+        var hasStyleableProperties = defined(styleableProperties);
+
+        // Use per-point normals to hide back-facing points.
+        var backFaceCulling = true;
+
+        var positionAttributeLocation = 0;
+        var colorAttributeLocation = 1;
+        var normalAttributeLocation = 2;
+        var batchIdAttributeLocation = 3;
+        var numberOfAttributes = 4;
+
+        var styleableShaderAttributes;
+        var styleableVertexAttributes;
+        var styleableVertexAttributeLocations;
+
+        if (hasStyleableProperties) {
+            styleableShaderAttributes = '';
+            styleableVertexAttributes = [];
+            styleableVertexAttributeLocations = {};
+
+            var attributeLocation = numberOfAttributes;
+
+            for (var name in styleableProperties) {
+                if (styleableProperties.hasOwnProperty(name)) {
+                    var property = styleableProperties[name];
+                    // TODO : this will not handle matrix types currently
```

Comment:
It could...and it likely will at some point...but I would rather roadmap this since it is not in the most promient use cases, update https://github.com/AnalyticalGraphicsInc/3d-tiles/issues/2